src/_pytest/warning_types.py

```diff
@@ -8,6 +8,8 @@ class PytestWarning(UserWarning):
     Base class for all warnings emitted by pytest.
     """
 
+    __module__ = "pytest"
+
```

Comment:
This cannot be used in the base class already I assume?

Reply:
yeah, `__module__` isn't inherited:

```pycon
>>> class C:
...     __module__ = 'hi'
...
>>> class D(C): pass
...
>>> D.__module__
'__main__'
>>> C.__module__
'hi'
```

We could do something with a metaclass though:

```pycon
>>> class mcs(type):
...     def __new__(cls, name, bases, dct):
...         ret = super().__new__(cls, name, bases, dct)
...         ret.__module__ = 'pytest'
...         return ret
...
>>> class C(Exception, metaclass=mcs): pass
...
>>> C.__module__
'pytest'
>>> class D(C): pass
...
>>> D.__module__
'pytest'
```

Reply:
Yeah, might be useful, but not sure if it is worth it - the test checks all classes from the module, so it would catch missing ones.

Reply:
the metaclass as shown would be actively hostile to plugin authors trying to inherit from a pytest warning

Reply:
Yep, as the test catches it, don't think it is worth it.

Reply:
oh for sure, I wasn't honestly suggesting metaclass magic 😆 -- just showing how it would be possible